### PR TITLE
fix issue  444

### PR DESCRIPTION
**Pull request by tomdong2010**

### CODE REVIEW

File: internal/qkcapi/input_args.go [hunk 1]
```diff
@@ -33,6 +34,11 @@ type CallArgs struct {
 	GasTokenID      *hexutil.Uint64  `json:"gasTokenId"`
 	TransferTokenID *hexutil.Uint64  `json:"transferTokenId"`
 }
+type GetAccount struct {
+	Address       *account.Address `json:"address"`
```
Comment (mask-pp) [inline internal/qkcapi/input_args.go]: account.Address don't need *

File: internal/qkcapi/api.go [hunk 2]
```diff
@@ -194,8 +194,11 @@ func (p *PublicBlockChainAPI) GetBalances(address account.Address, blockNr *rpc.
 	return fields, nil
 }
 
-func (p *PublicBlockChainAPI) GetAccountData(address account.Address, blockNr *rpc.BlockNumber, includeShards *bool) (map[string]interface{}, error) {
-	if includeShards != nil && blockNr == nil {
+func (p *PublicBlockChainAPI) GetAccountData(getAccount GetAccount) (map[string]interface{}, error) {
+	address := account.Address{getAccount.Address.Recipient, getAccount.Address.FullShardKey}
```
Comment (mask-pp) [inline internal/qkcapi/api.go]: attention format

File: internal/qkcapi/input_args.go (hunk 1 above)
Comment (tomdong2010) [reply internal/qkcapi/input_args.go]: ok

File: internal/qkcapi/input_args.go [hunk 3]
```diff
@@ -33,6 +34,11 @@ type CallArgs struct {
 	GasTokenID      *hexutil.Uint64  `json:"gasTokenId"`
 	TransferTokenID *hexutil.Uint64  `json:"transferTokenId"`
 }
+type GetAccount struct {
```
Comment (pingke) [inline internal/qkcapi/input_args.go]: type GetAccountDataArgs struct {
	Address       account.Address `json:"address"`
	IncludeShards *bool           `json:"include_shards"`
	BlockNr       *int64          `json:"block_nr"`
}

File: internal/qkcapi/api.go (hunk 2 above)
Comment (pingke) [inline internal/qkcapi/api.go]: func (p *PublicBlockChainAPI) GetAccountData(args GetAccountDataArgs) (map[string]interface{}, error) {
	address, blockNr, includeShards := args.Address, getAccount.BlockNr, getAccount.IncludeShards

Comment (tomdong2010) [reply internal/qkcapi/api.go]: ok

File: internal/qkcapi/input_args.go (hunk 3 above)
Comment (tomdong2010) [reply internal/qkcapi/input_args.go]: ok

File: cluster/master/handle.go [hunk 4]
```diff
@@ -624,7 +624,9 @@ func (pm *ProtocolManager) HandleNewTransactionListRequest(peerId string, rpcId
 				//TODO need err
 			}
 			if hashList == nil {
-				hashList = result.Hashes
+				if result != nil {
```
[the remark this answers is not in the document]
Comment (tomdong2010) [reply cluster/master/handle.go]: ok

File: internal/qkcapi/api.go (hunk 2 above)
Comment (tomdong2010) [reply internal/qkcapi/api.go]: ok

**Comment by tomdong2010** [comment]
after  change,i test  , "getAccountData" api ，it passed in quarkchain-dev.  i curl it ,map can pass ,list not pass ,this look like python

**Comment by tomdong2010** [comment]
image

File: internal/qkcapi/input_args.go [hunk 5]
```diff
@@ -34,6 +34,13 @@ type CallArgs struct {
 	TransferTokenID *hexutil.Uint64  `json:"transferTokenId"`
 }
 
+
+type GetAccountDataArgs struct {
+	Address        account.Address   `json:"address"`
+	IncludeShards  *bool             `json:"include_shards"`
+	BlockNr         *rpc.BlockNumber            `json:"block_nr"`
```
Comment (pingke) [inline internal/qkcapi/input_args.go]: format

File: internal/qkcapi/api.go [hunk 6]
```diff
@@ -194,8 +194,9 @@ func (p *PublicBlockChainAPI) GetBalances(address account.Address, blockNr *rpc.
 	return fields, nil
 }
 
-func (p *PublicBlockChainAPI) GetAccountData(address account.Address, blockNr *rpc.BlockNumber, includeShards *bool) (map[string]interface{}, error) {
-	if includeShards != nil && blockNr == nil {
+func (p *PublicBlockChainAPI) GetAccountData(getAccount GetAccountDataArgs) (map[string]interface{}, error) {
```
Comment (pingke) [inline internal/qkcapi/api.go]: getAccount -> args ?

**Pull request merged**